inc/3rd-party/amazon-s3-and-cloudfront/inc/classes/class-imagify-as3cf-attachment.php

```diff
@@ -335,6 +335,178 @@ public function optimize( $optimization_level = null, $metadata = array() ) {
 	}
 
 	/**
+	 * Optimize missing thumbnail sizes with Imagify.
+	 *
+	 * @since  1.6.10
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @param  int $optimization_level The optimization level (2=ultra, 1=aggressive, 0=normal).
+	 * @return array|object            An array of thumbnail data, size by size. A WP_Error object on failure.
+	 */
+	public function optimize_missing_thumbnails( $optimization_level = null ) {
+		// Check if the attachment extension is allowed.
+		if ( ! $this->is_mime_type_supported() ) {
+			return array();
```

Comment:
Actually I put this test only to be safe, because it wouldn't be possible to go through this test.
But that's not a bad idea, I'll add it because it makes more sense.